code/modelCuration/getCompFromUniprotCellAtlas.py

```diff
@@ -0,0 +1,138 @@
+# 'Microtubules', 'Midbody ring', 'Midbody', 'Cytokinetic bridge', 'Microtubule ends', 'Mitotic spindle'
+# 'Intermediate filaments', 'Actin filaments', 'Focal adhesion sites', 'Cleavage furrow', 
+# 'Lipid droplets', 'Vesicles', 'Endosomes', 'Plasma membrane', 'Cell Junctions'
+# 'Centrosome', 'Centriolar satellite'
+
+# construct a dict for key word mapping to Cell Atlas compartments
+cellatlas_keywords = {
+    #'Extracellular': [''],   # e
+    'Peroxisome': ['Peroxisomes'],   # x
+    'Mitochondria': ['Mitochondria'],   # m
+    'Cytosol': ['Cytosol', 'Cytoplasmic bodies', 'Rods & Rings', 'Aggresome'],   # c
+    'Lysosome': ['Lysosomes'],   # l
+    'Endoplasmic reticulum': ['Endoplasmic reticulum'],   # r
+    'Golgi apparatus': ['Golgi apparatus'],   # g
+    'Nucleus': ['Nuclear membrane', 'Nucleoli rim', 'Nucleoli', 'Nuclear bodies', 'Nucleoli fibrillar center', 'Nucleoplasm', 'Kinetochore', 'Mitotic chromosome', 'Nuclear speckles'],   # n
+    #'Inner mitochondria': ['']    # i, Cell Atlas does provide such location info
+}
+
+
+# store CellAtlas compartment info to a dict with keys as ensembl ids
+geneComps_from_cell_atlas = {}
+for gene in Human_genes:
+    out_list = []  # store compartments as a list
+    if gene not in cellAtlas_ensembl_id:
+        geneComps_from_cell_atlas[gene] = out_list
+    else:
+        gene_ind = cellAtlas_ensembl_id.index(gene)
+        for key in cellatlas_keywords:
+            # check if cellular location text contains any keyword from a compartment
+            if any(word in cellAtlas_comps[gene_ind] for word in cellatlas_keywords[key]):
+                out_list.append(key)
+        geneComps_from_cell_atlas[gene] = out_list
+
+
+# integrate compartment info from cellAtlas and SwissProt with following rules:
+# 1. output two columns: "compartments" and "compDataSource"
+# 2. use one source if another has no assigned compartments
+# 3. union compartments if both source are provided
+swissprot_comps = pd.read_table("../../data/modelCuration/Swissprot_compartments.tsv")
+geneComps_from_swissprot = swissprot_comps['compartments'].to_list()
+
+# combine compartment info from two data sources
+geneComps_combined = []
+source = ['']*len(Human_genes)
+for i, gene in enumerate(Human_genes):
+    out_list = []  # save compartmetns as a list
+    if pd.isna(geneComps_from_swissprot[i]) and geneComps_from_cell_atlas[gene] != []:
+        out_list = geneComps_from_cell_atlas[gene]
+        source[i] = 'CellAtlas'
+    elif not pd.isna(geneComps_from_swissprot[i]) and geneComps_from_cell_atlas[gene] == []:
+        out_list = geneComps_from_swissprot[i].split(';')
+        source[i] = 'SwissProt'
+    elif not pd.isna(geneComps_from_swissprot[i]) and geneComps_from_cell_atlas[gene] != []:
+        union = set(geneComps_from_swissprot[i].split(';')+ geneComps_from_cell_atlas[gene])
+        out_list = list(union)
+        source[i] = 'SwissProt&CellAtlas'
```

Comment:
For cases when `source` is more than one, maybe use `;` as the delimiter instead of `&`, since we typically use the former as a delimiter.